worlds/banjo_tooie/test/test_signpost_hints.py

```diff
@@ -0,0 +1,191 @@
+            possible_moves += 16
+
+        assert move_hints >= min(self.world.options.signpost_move_hints, possible_moves,
+                                 self.world.options.signpost_hints)
+
+
+class TestClearSignpostsHints(TestSignpostsHints):
+    options = {
+        "hint_clarity": HintClarity.option_clear
+    }
+
+    def test_accurate_hint_text(self) -> None:
+        for hint_data in self.world.hints.values():
+            if not hint_data.location_id:
+                continue
+            hinted_location = [
+                location for location in self.world.get_locations()
+                if location.address == hint_data.location_id
+            ][0]
+
+            text = hint_data.text
+
+            location = hinted_location.name
+            item = hinted_location.item.name
+
+            assert 'Your' in text
+            assert location in text
+            assert 'your' in text
+            assert item in text
+
+
+class TestClearSignpostsHintsAddAllHints(TestSignpostsHints):
+    options = {
+        "hint_clarity": HintClarity.option_clear,
+        "add_signpost_hints_to_ap": AddSignpostHintsToArchipelagoHints.option_always
+    }
+
+    def test_should_add_hint(self) -> None:
+        for hint_data in self.world.hints.values():
+            if not hint_data.location_id:
+                assert not hint_data.should_add_hint
+            else:
+                assert hint_data.should_add_hint
+
+
+class TestClearSignpostsHintsAddAllProgressionHints(TestSignpostsHints):
+    options = {
+        "hint_clarity": HintClarity.option_clear,
+        "add_signpost_hints_to_ap": AddSignpostHintsToArchipelagoHints.option_progression
+    }
+
+    def test_should_add_hint(self) -> None:
+        for hint_data in self.world.hints.values():
+            if not hint_data.location_id:
+                assert not hint_data.should_add_hint
+                continue
+            hinted_location = [
+                location for location in self.world.get_locations()
+                if location.address == hint_data.location_id
+            ][0]
+
+            if hinted_location.item.advancement:
+                assert hint_data.should_add_hint
+            else:
+                assert not hint_data.should_add_hint
+
+
+class TestCrypticSignpostsHints(TestSignpostsHints):
+    options = {
+        "hint_clarity": HintClarity.option_cryptic,
+        "add_signpost_hints_to_ap": AddSignpostHintsToArchipelagoHints.option_always
+    }
+
+    def test_accurate_hint_text(self) -> None:
+        for hint_data in self.world.hints.values():
+            if not hint_data.location_id:
+                continue
+            hinted_location = [
+                location for location in self.world.get_locations()
+                if location.address == hint_data.location_id
+                and location.player == hint_data.location_player_id
+            ][0]
+
+            text = hint_data.text[::]
```

Comment:
Is there a reason to not just use `hint_data.text` below (or remove the copy through slicing here)?